### PR TITLE
fixes oversight with organ removal proc

### LINKED ISSUE #15208: Plasmaman had infinite lungs
(WEB REPORT BY: moltevolte REMOTE: 172.93.109.202:7777)
>General description of the isse
I removed 10+ lungs from a plasmamen.

>What you expected to happen
The lungs to be removed normally so that the new one can be inserted.

>What actually happened
When I tried to remove and replace plasmaman's lungs I couldn't put new ones in, because he already had lungs. I tried to remove them again, it worked. Then I repeated it a few more times, every time new lungs plopped out of the guy. I stopped after 10 tries, but I guess that I could have done it indefinitely.

>Steps to reproduce if possible
Try to replace burst lungs on a plasmamen.

### PR DESCRIPTION
**Pull request by ghost**
Previously it was going off of the name of the organ, so special species organs (Such as plasmaman lungs, vox lungs, grey eyes) would not get properly dereferenced from internal_organs_by_name

Now, internal organs have an organ_type variable that stores the string of what type of organ they are you can refer to, rather than having to magic the string from somewhere

fixes #15208

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1678,35 +1678,39 @@ mob/living/carbon/human/isincrit()
...
 
-/mob/living/carbon/human/get_heart()	
+/mob/living/carbon/human/get_heart()
 	return internal_organs_by_name["heart"]
-	
+
 //Moved from internal organ surgery
 //Removes organ from src, places organ object under user
+//Needs to be passed the name of the TYPE of organ with no capitalization, else it fails. No /improver Vox Lungs please
 //example: H.remove_internal_organ(H,internal_organs_by_name["heart"],H.get_organ(LIMB_CHEST))
```
Comment (SonixApache) [inline code/modules/mob/living/carbon/human/human.dm]: update example

Comment (SonixApache) [inline code/modules/mob/living/carbon/human/human.dm]: Update example.

**Comment by SonixApache** [comment]
i for one blame the xenos

**Comment by Intigracy** [comment]
internal_organs_by_name["heart"] should just return the correct type of heart they have instead of having to be passed the specific strings

and etc for the other organs

**Comment by Shadowmech88** [comment]
`remove_internal_organ()` is used by heart attack syndrome, make sure you fix that instance as well now that it's been merged.

**Comment by ghost** [comment]
It's a referential list where the memory address is assigned to a string, I would assume it would need to be passed that string to work at all, and the previous way of acquiring that string is not possible due to the organ names not fitting with said string.

Maybe each internal organ could have a string of their organ type? Would that help improve things?

@Shadowmech88 good call

**Comment by ghost** [comment]
there we go, now rather than having to pass a string to the proc, the proc gets the string from the organ itself.

Should be useful in the future, and mean that no current uses of the proc need to be editted.

**Pull request merged**